src/Land/Model/SurfaceFlow.jl

```diff
@@ -0,0 +1,256 @@
+#end
+
+"""
+    flux(::SurfaceWaterArea, ::VolumeAdvection, land::LandModel, args,)
+
+A first order flux method for the OverlandFlow model, adding in advection of water volume due to flowing water.
+"""
+function flux(::SurfaceWaterArea, ::VolumeAdvection, land::LandModel, args)
+    @unpack state, aux = args
+    x = aux.x
+    y = aux.y
+    width = land.surface.width(x, y)
+    area = max(eltype(state)(0.0), state.surface.area)
+    height = area / width
+    v = calculate_velocity(land.surface, x, y, height)
+    Q = area * v
+    return Q
+end
+#function flux_first_order!(land::LandModel, surface::OverlandFlowModel, fl#ux::Grad, state::Vars, aux::Vars, t::Real, directions)
+#    x = aux.x
+#    y = aux.y
+#    width = surface.width(x, y)
+#    area = max(eltype(state)(0.0), state.surface.area)
+#    height = area / width
+#    v = calculate_velocity(surface, x, y, height)
+#    Q = area * v
+#    flux.surface.area = Q
+#end 
+
+# Boundary Conditions
+
+# General case - to be used with bc::NoBC
+function surface_boundary_flux!(
+    nf,
+    bc::Land.AbstractBoundaryConditions,
+    m::Union{NoSurfaceFlowModel, OverlandFlowModel},
+    land::LandModel,
+    _...,
+) end
+
+function surface_boundary_state!(
+    nf,
+    bc::Land.AbstractBoundaryConditions,
+    m::Union{NoSurfaceFlowModel, OverlandFlowModel},
+    land::LandModel,
+    _...,
+) end
+
+"""
+    surface_boundary_flux!(
+        nf,
+        bc::Land.Dirichlet,
+        model::OverlandFlowModel,
+        land::LandModel,
+        _...,
+    )
+
+The surface boundary flux function for the OverlandFlow model, which
+does nothing if Dirichlet conditions are chosen in order to not
+overconstrain the solution.
+"""
+function surface_boundary_flux!(
+    nf,
+    bc::Land.Dirichlet,
+    model::OverlandFlowModel,
+    land::LandModel,
+    _...,
+) end
+
+"""
+    surface_boundary_state!(
+        nf,
+        bc::Land.Dirichlet,
+        model::OverlandFlowModel,
+        land::LandModel,
+        state⁺::Vars,
+        aux⁺::Vars,
+        nM,
+        state⁻::Vars,
+        aux⁻::Vars,
+        t,
+        _...,
+    )
+
+The surface boundary state function for the OverlandFlow model when
+Dirichlet conditions are chosen.
+"""
+function surface_boundary_state!(
+    nf,
+    bc::Land.Dirichlet,
+    model::OverlandFlowModel,
+    land::LandModel,
+    state⁺::Vars,
+    aux⁺::Vars,
+    nM,
+    state⁻::Vars,
+    aux⁻::Vars,
+    t,
+    _...,
+)
+    bc_function = bc.state_bc
+    state⁺.surface.area = bc_function(aux⁻, t)
+end
+
+"""
+    Precip <: TendencyDef{Source}
+
+A source term for overland flow where a prescribed net precipitation
+drives the overland flow.
+"""
+struct Precip{FT, F} <: TendencyDef{Source}
+    precip::F
+
+    function Precip{FT}(precip::F) where {FT, F}
+        new{FT, F}(precip)
+    end
+end
+
+function (p::Precip{FT})(x, y, t) where {FT}
+    FT(p.precip(x, y, t))
+end
+
+
+prognostic_vars(::Precip) = (SurfaceWaterArea(),)
```

Comment:
We need this definition to provide information as to which prognostic variables each source should be applied to only because sources are passed in as a `Tuple` at the driver level. Fluxes are handled differently since they're all handled in `eq_tends` and not passed in from the driver.